src/app/services/association-injector/association-injector.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable, Injector } from "@angular/core";
+import { BAW_SERVICE_OPTIONS } from "../baw-api/api-common";
+import { BawApiService } from "../baw-api/baw-api.service";
+import {
+  services,
+  serviceTokens,
+  serviceResolvers,
+} from "../baw-api/ServiceProviders";
+
+//! WARNING: to prevent a circular dependency issue, this service should be imported
+// through its "ASSOCIATION_INJECTOR" token
+@Injectable()
+export class AssociationInjectorService {
+  public constructor(private injector: Injector) {}
+
+  public get instance(): Injector {
+    this._instance ??= this.createInstance();
+    return this._instance;
+  }
+
+  private _instance?: Injector;
+
+  private createInstance(): Injector {
+    return Injector.create({
+      name: "AssociationInjector",
+      parent: this.injector,
+      providers: [
+        { provide: BawApiService, useClass: BawApiService },
+        {
+          provide: BAW_SERVICE_OPTIONS,
+          useValue: { disableNotification: true },
+        },
+        ...services,
+        ...serviceTokens,
+        ...serviceResolvers,
+      ],
+    });
+  }
+}
```

Comment:
why is this a service?

Can't it, itself just be a useFactory?

```
{
  provide: Injector // or AssociationInjector if we subclass injector
  useFactory: (injector) => { 
     // rest of the owl
     Injector.create( // ...
   },
   deps: [Injector]
}
```

Reply:
I've implemented this change. It does make providing the `ASSOCIATION_INJECTOR` slightly more verbose

e.g.
I now have to repeat this code whenever I want to provide the `AssociationInjector`

![image](https://github.com/user-attachments/assets/798ff5a5-5035-459d-ba2e-b0f1e2257923)